apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -272,6 +272,12 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return packageDescription;
   }
 
+  private _getValidDependencyVersion(dependencyVersion: string): string {
+    return semver.valid(dependencyVersion.split('/').pop()!) ?
```

Comment:
I don't think this is what you want. From that issue, it looks like the two formats we need to handle are:
- `/react-loadable/5.5.0/react@16.8.6`
- `/react-loadable/5.5.0_react@16.8.6`

(both matching `react@16.8.6`)

The `dependencyVersion.split('/').pop()` statement will return `react@16.8.6`, but `semver.valid('react@16.8.6')` returns `null`, always falling back to the `dependencyVersion.split('_')` statement.

I think this is what we want:

```suggestion
    if (dependencyVersion.indexOf('_') !== -1) {
      // PNPM 3: the format looks like "/react-loadable/5.5.0_react@16.8.6"
      return dependencyVersion.split('_').pop()!;
    } else {
      // PNPM 2: the format looks like "/react-loadable/5.5.0/react@16.8.6"
      return dependencyVersion.split('/').pop()!;
    }    
```

Reply:
@iclanton in PNP 2.x it seems like its coming in as:
/react-loadable/5.5.0 and
/react-loadable/5.5.0/react@16.8.6
and similarly for react-dom. So  `semver.valid('5.5.0')` or `semver.valid('16.8.6')` passes 

In pnpm 3.x it is consistently:
/react-loadable/5.5.0_react@16.8.6
/react-dom/16.8.6_react@16.8.6
and `dependencyVersion.split('_')[0]` will validate `semver.valid('5.5.0')` 

If we use `dependencyVersion.split('_').pop()!;` it will fail in pnpm 3.x because it will try to validate `react@16.8.6` 

I think we need to return `dependencyVersion.split('_')[0]` for pnpm 3.x to pass regardless if we check using `dependencyVersion.indexOf('_') !== -1` or `semver.valid`

Reply:
I misunderstood what the input to this function was. I thought it was the full path, not just a portion of the version string.

None of the inputs to this function when it runs on WBT contain any slashes. That makes me wonder if the logic introduced in https://github.com/microsoft/web-build-tools/pull/615 is necessary for anything other than PNPM v1. @halfnibble is probably right that `.split('_')` should be called after the slashes are split off.